platform/linux-generic/pktio/dpdk.c

```diff
@@ -587,9 +587,12 @@ static inline int mbuf_to_pkt(pktio_entry_t *pktio_entry,
 	pkt_dpdk_t *pkt_dpdk = pkt_priv(pktio_entry);
 	odp_pool_t pool = pkt_dpdk->pool;
 	odp_pktin_config_opt_t pktin_cfg = pktio_entry->s.config.pktin;
-	odp_proto_layer_t parse_layer = pktio_entry->s.config.parser.layer;
 	odp_pktio_t input = pktio_entry->s.handle;
 	uint16_t frame_offset = pktio_entry->s.pktin_frame_offset;
+	const odp_proto_layer_t layer =
+		pktio_cls_enabled(pktio_entry) ?
+			ODP_PROTO_LAYER_ALL :
+			pktio_entry->s.config.parser.layer;
```

Comment:
This value could be saved to `pktio_entry_t` to reduce overhead.

Reply:
v4.